include/arch/arc/arch.h

```diff
@@ -70,7 +70,7 @@ extern "C" {
 
 #if defined(CONFIG_MPU_STACK_GUARD)
 	#if CONFIG_ARC_MPU_VER == 2
-	#define STACK_GUARD_SIZE 2048
+	#define STACK_GUARD_SIZE 0
```

Comment:
This combination isn't supported, looks like these complex nested #ifdef's can be simplified.

Reply:
already done